### ISSUE TITLE
Debian-based images need git to build Spigot from source

### ISSUE THREAD

**Issue by chewi** [itzg/docker-minecraft-server#819]
I'm trying to run this in Azure but [this bizarre issue](https://gitlab.alpinelinux.org/alpine/aports/-/issues/10960) is causing problems with the Alpine-based images so I'm trying Debian instead.

That's not working either though because it's missing git, which is needed to build Spigot from source.

```
[init] Changing ownership of /data to 1000 ...
[init] Running as uid=1000 gid=1000 with /data as 'drwxrwxrwx 2 0 0 0 Mar 30 09:36 /data'
[init] Resolved version given LATEST into 1.16.5
[init] Resolving type given SPIGOT
[init] Building Spigot 1.16.5 from source, might take a while, get some coffee
[init] ....[init] done
mv: cannot stat 'spigot-*.jar': No such file or directory
[init] ERR failed to build Spigot
Loading BuildTools version: git-BuildTools-36c11e9-126 (#126)
Java Version: Java 11
Current Path: /data/temp/.
Could not successfully run git. Please ensure it is installed and functioning. Cannot run program "git" (in directory "."): error=2, No such file or directory
```

**Comment by chewi** [comment]
I've worked around this by installing git in the init command before running `/start` and it works... well kinda, the build is still broken but that's a different problem!

**Comment by itzg** [comment]
I'm curious why you want to build Spigot and not just use the prebuilt download? The build process requires several extra tools (such as needing to add git), takes a really long time to build, and as you have found is brittle. So I have been considering removing the feature entirely.

**Comment by chewi** [comment]
The prebuilt version is technically illegal. I do try to stay on the right side of licensing laws and would especially like to in this case, given who I work for. :wink: Please don't remove the feature.

git seems to be the only thing missing. I think the failure I've now seen is somehow related to the Azure environment and I will try to figure it out.

**Comment by itzg** [comment]
That's a totally fair point. 

Which of the debian image tags are you using specifically and I'll ensure I get it merged to that branch/tag for your testing?

**Comment by chewi** [comment]
Thanks! It's multiarch.

**Comment by itzg** [comment]
The `itzg/minecraft-server:multiarch` image now includes git.

**Comment by chewi** [comment]
Awesome, thanks again!

Just before your update, I actually did manage to get it working with my hacked version. I'm still not sure exactly what the problem was though. It had been failing to apply the Spigot patches, stopping on a random patch like this:

```
Applying: Add creative mode NBT permissions
error: Your local changes to the following files would be overwritten by merge:
	src/main/java/net/minecraft/world/item/ItemDebugStick.java
Please commit your changes or stash them before you merge.
Aborting
```

When I tried to reproduce it manually, I wasn't able to but it was taking an age to do anything so I recreated the storage using Azure's Premium tier rather than its cheaper Transaction-optimised tier. Turns out the former isn't quite as expensive as I thought and I get more than enough free credit anyway! On the Premium tier, it just worked first time. I wouldn't expect it to break simply due to being slower so maybe there are subtle concurrency differences between the tiers that confuse git.

So I wasn't able to test your change by building Spigot but I can see from execing into the container that git is now already present so it should be fine now.
